### PR TITLE
added 'sync move' function and example and some formatting

### PR DESCRIPTION
**Pull request by alireza7575**
This pull request introduces a new 'sync move' function to the codebase, along with an accompanying usage example.

The 'sync move' function enhances our library by allowing users to synchronize the movement of multiple servos. This is particularly beneficial in scenarios where coordinated movement of servos is required.

The usage example provides a clear demonstration of how to use the 'sync move' function, which will aid users in correctly implementing this function in their projects.

Changes included:

1. Addition of the 'sync move' function in STSServoDriver.cpp and STSServoDriver.h.
2. Addition of a 'sync move' usage example in examples/syncMove/syncMove.ino.

I encourage reviewers to test the function using the provided example and offer feedback on the implementation. Any suggestions for improvements or optimizations are always welcome.

Thank you.

### CODE REVIEW

**Comment by matthieuvigne** [comment]
Sorry for not responding earlier, I had github notification ill-configured and didn't see the PR. But thank you for developing this, this is certainly a nice addition!

One comment, I noticed you changed identation from 4 spaces to tab. Could you revert this ? I like it better with spaces, simply because it's better supported across IDEs ; and as is is creates a diff on basically every line of the code ;)

Other than that, I'm fine with merging this - or I can wait if you are still developing, I see you've added several commits in the past few months.

File: examples/SyncMove/SyncMove.ino [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Basic example of making a STS3215 servos move async.
```
Comment (matthieuvigne) [inline examples/SyncMove/SyncMove.ino]: rather *two servos* move *synchronously*   - async is already all we had before.

Comment (alireza7575) [reply examples/SyncMove/SyncMove.ino]: Thanks for your review and sorry for the formatting. I already fixed it. Moreover, to make it easier I added several functionalities such as setTargetAcceleration, setPositionOffset, setTargetPositions, and added speed to setTargetPosition. currently, that's all of the changes.

Comment (matthieuvigne) [reply examples/SyncMove/SyncMove.ino]: Great, I'll merge this and make a new release.

Comment (alireza7575) [reply examples/SyncMove/SyncMove.ino]: Great thanks.

**Pull request merged**